graveler/graveler_test.go

```diff
@@ -429,6 +429,142 @@ func TestGraveler_Commit(t *testing.T) {
 	}
 }
 
+func TestGraveler_CommitExistingTree(t *testing.T) {
+	expectedCommitID := graveler.CommitID("expectedCommitId")
+	expectedTreeID := graveler.TreeID("expectedTreeID")
+	type fields struct {
+		CommittedManager *testutil.CommittedFake
+		StagingManager   *testutil.StagingFake
+		RefManager       *testutil.RefsFake
+	}
+	type args struct {
+		ctx          context.Context
+		repositoryID graveler.RepositoryID
+		branchID     graveler.BranchID
+		committer    string
+		message      string
+		metadata     graveler.Metadata
+	}
+	tests := []struct {
+		name        string
+		fields      fields
+		args        args
+		want        graveler.CommitID
+		expectedErr error
+	}{
+		{
+			name: "valid existing commit",
+			fields: fields{
+				CommittedManager: &testutil.CommittedFake{TreeID: expectedTreeID, ValueIterator: testutil.NewValueIteratorFake([]graveler.ValueRecord{})},
+				StagingManager:   &testutil.StagingFake{ValueIterator: testutil.NewValueIteratorFake([]graveler.ValueRecord{})},
+				RefManager:       &testutil.RefsFake{CommitID: expectedCommitID, Branch: &graveler.Branch{CommitID: expectedCommitID}, Commit: &graveler.Commit{TreeID: expectedTreeID}},
+			},
+			args: args{
+				ctx:          nil,
+				repositoryID: "repo",
+				branchID:     "branch",
+				committer:    "committer",
+				message:      "a message",
+				metadata:     graveler.Metadata{},
+			},
+			want:        expectedCommitID,
+			expectedErr: nil,
+		},
+		{
+			name: "staging not empty",
+			fields: fields{
+				CommittedManager: &testutil.CommittedFake{TreeID: expectedTreeID},
+				StagingManager:   &testutil.StagingFake{ValueIterator: testutil.NewValueIteratorFake([]graveler.ValueRecord{{Key: graveler.Key("key"), Value: &graveler.Value{}}})},
+				RefManager:       &testutil.RefsFake{CommitID: expectedCommitID, Branch: &graveler.Branch{CommitID: expectedCommitID}, Commit: &graveler.Commit{TreeID: expectedTreeID}},
+			},
+			args: args{
+				ctx:          nil,
+				repositoryID: "repo",
+				branchID:     "branch",
+				committer:    "committer",
+				message:      "a message",
+				metadata:     nil,
+			},
+			want:        expectedCommitID,
+			expectedErr: graveler.ErrDirtyBranch,
+		},
+		{
+			name: "tree not found",
+			fields: fields{
+				CommittedManager: &testutil.CommittedFake{Err: graveler.ErrNotFound},
+				StagingManager:   &testutil.StagingFake{ValueIterator: testutil.NewValueIteratorFake([]graveler.ValueRecord{})},
+				RefManager:       &testutil.RefsFake{CommitID: expectedCommitID, Branch: &graveler.Branch{CommitID: expectedCommitID}, Commit: &graveler.Commit{TreeID: expectedTreeID}},
+			},
+			args: args{
+				ctx:          nil,
+				repositoryID: "repo",
+				branchID:     "branch",
+				committer:    "committer",
+				message:      "a message",
+				metadata:     nil,
+			},
+			want:        expectedCommitID,
+			expectedErr: graveler.ErrTreeNotFound,
+		},
+		{
```

Comment:
do we also need "branch not found"?

Reply:
Harder with the current fakes and adds little coverage.
Deferring until we'll switch to generated mocks here too.